src/fileformat/file_format/pe/authenticode/x509_certificate.cpp

```diff
@@ -24,18 +24,60 @@ X509Certificate::X509Certificate(const X509* cert)
 
 std::string X509Certificate::getSerialNumber() const
 {
-	ASN1_INTEGER* serial_number_asn1 = X509_get_serialNumber(const_cast<X509*>(cert));
-	BIGNUM* bignum = ASN1_INTEGER_to_BN(serial_number_asn1, nullptr);
-
-	BIO* bio = BIO_new(BIO_s_mem());
-	BN_print(bio, bignum);
-	auto data_len = BIO_number_written(bio);
-
-	std::vector<char> result(data_len);
-	BIO_read(bio, static_cast<void*>(result.data()), data_len);
-	BIO_free_all(bio);
-	BN_free(bignum);
-	return { result.begin(), result.end() };
+    // https://github.com/VirusTotal/yara/blob/879a6576dd6e544bf9fc7711821029bf842fac54/libyara/modules/pe/pe.c#L1316
+    ASN1_INTEGER* serial_number_asn1 = X509_get_serialNumber(const_cast<X509*>(cert));
+    if (serial_number_asn1) {
```

Comment:
This is just a nitpick but would it be possible to rather do early return from the function, than having several tens of lines of codes indented like this? What I mean by that is that instead of

```cpp
ASN1_INTEGER* serial_number_asn1 = X509_get_serialNumber(const_cast<X509*>(cert));
if (serial_number_asn1) {
    // ...
}

return {};
```

you'd have

```cpp
ASN1_INTEGER* serial_number_asn1 = X509_get_serialNumber(const_cast<X509*>(cert));
if (!serial_number_asn1)
    return {};

// ...
```

Reply:
Changed as requested